pennylane/ops/qutrit/non_parametric_ops.py

```diff
@@ -430,3 +430,163 @@ def pow(self, z):
 
     def adjoint(self):
         return TSWAP(wires=self.wires)
+
+
+class THadamard(Operation):
+    r"""THadamard(wires, subspace)
+    The ternary Hadamard operator
+
+    Performs the Hadamard operation on the specified 2D subspace if specified. The subspace is
+    given as a keyword argument and determines which two of three single-qutrit basis states the
+    operation applies to. When a subspace is not specified, the generalized Hadamard operation
+    is used.
+
+    The construction of this operator is based on section 2 of
+    `Di et al. (2012) <https://arxiv.org/abs/1105.5485>`_ when the subspace is specified, and
+    definition 4 and equation 5 from `Yeh et al. (2022) <https://arxiv.org/abs/2204.00552>`_
+    when no subspace is specified.
+
+    **Details:**
+
+    * Number of wires: 1
+    * Number of parameters: 0
+
+    Args:
+        wires (Sequence[int] or int): the wire the operation acts on
+        subspace (Sequence[int]): the 2D subspace on which to apply operation. This should be
+            `None` for the generalized Hadamard.
+        do_queue (bool): Indicates whether the operator should be
+            immediately pushed into the Operator queue (optional)
+
+    **Example**
```

Comment:
@KetpuntoG tagging you here to give you a sense of what this operation is going to look like (both the subspace and non-subspace version that you need, which is the last example here).